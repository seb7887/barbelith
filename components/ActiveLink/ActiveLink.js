import { withRouter } from 'next/router';

const defaultRouter = {
  pathname: '/',
  asPath: '/',
  push: () => { },
  prefetch: () => { }
};

const ActiveLink = ({ router = defaultRouter, href, children }) => {
  // IIFE function that prefetchs the page requested by the link
  (function prefetchPages() {
    if (typeof window !== 'undefined') {
      router.prefetch(router.pathname);
    }
  })();

  const handleClick = e => {
    e.preventDefault();
    router.push(href);
  };

  const isCurrentPath = router.pathname === href || router.asPath === href;

  return (
    <div data-testid='active-link'>
      <a
        href={href}
        onClick={handleClick}
        style={{
          textDecoration: 'none',
          margin: 0,
          padding: 0,
          fontWeight: isCurrentPath ? 'bold' : 'normal',
          color: '#000',
          transition: 'all 0.3s'
        }}
      >
        {children}
      </a>
    </div>
  );
};

export default withRouter(ActiveLink);
