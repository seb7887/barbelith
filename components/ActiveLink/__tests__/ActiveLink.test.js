/* eslint-env jest */
import { cleanup, render } from 'react-testing-library';
import Router from 'next/router';
import ActiveLink from '../ActiveLink';

Router.router = {
  push() { },
  prefetch() { }
};

const mockedChildren = 'Barbelith';

afterEach(cleanup);

describe('<ActiveLink/>', () => {
  it('renders and matches snapshot', () => {
    const { getByTestId, asFragment } = render(
      <ActiveLink href='/' {...Router}>
        {mockedChildren}
      </ActiveLink>
    );

    expect(getByTestId('active-link').textContent).toContain('Barbelith');
    expect(asFragment()).toMatchSnapshot();
  });
});