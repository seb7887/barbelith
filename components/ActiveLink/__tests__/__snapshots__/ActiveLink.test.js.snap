// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveLink/> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="active-link"
  >
    <a
      href="/"
      style="text-decoration: none; margin: 0px; padding: 0px; font-weight: normal; color: rgb(0, 0, 0);"
    >
      Barbelith
    </a>
  </div>
</DocumentFragment>
`;
